### PR TITLE
F/bastion hosts

### PR DESCRIPTION
**Pull request by sa-progress**
### Description

Please describe what this change achieves. Ensure you have read the [Contributing to InSpec](https://github.com/inspec/inspec-azure/CONTRIBUTING.MD) document before submitting.

### Issues Resolved

List any existing issues this PR resolves, or any Discourse or StackOverflow discussion that's relevant

### Check List

- [x] New functionality includes tests
- [x] All tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: libraries/azure_bastion_hosts_resources.rb [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    # This will ensure constructing resource and passing `should_not exist` test.
+    return [] if @resources.empty?
+
+    @resources.each do |resource|
+      @table << {
+        id: resource[:id],
+        name: resource[:name],
+        type: resource[:type],
+        tags: resource[:tags],
+        provisioningState: resource[:properties][:provisioningState],
```
Comment (sathish-progress) [inline libraries/azure_bastion_hosts_resources.rb:91]: do we not need other properties?

File: libraries/backend/azure_environment.rb [hunk 2]
```diff
@@ -98,8 +98,8 @@ def initialize(options)
                                                            active_directory_graph_api_version: '2013-04-05',
                                                            storage_endpoint_suffix: '.core.windows.net',
                                                            key_vault_dns_suffix: '.vault.azure.net',
-                                                           datalake_store_filesystem_endpoint_suffix: 'azuredatalakestore.net',
-                                                           datalake_analytics_catalog_and_job_endpoint_suffix: 'azuredatalakeanalytics.net',
+                                                           datalake_store_filesystem_endpoint_suffix: '.azuredatalakestore.net',
```
Comment (sathish-progress) [inline libraries/backend/azure_environment.rb:101]: please remove these changes..

File: README.md [hunk 3]
```diff
@@ -391,7 +392,9 @@ They can be defined as environment variables or resource parameters (has priorit
 | azurerm_virtual_machine_disk, azurerm_virtual_machine_disks | `2017-03-30` | [azure_virtual_machine_disk](docs/resources/azure_virtual_machine_disk.md), [azure_virtual_machine_disks](docs/resources/azure_virtual_machine_disks.md) |
 | azurerm_virtual_network, azurerm_virtual_networks | `2018-02-01` | [azure_virtual_network](docs/resources/azure_virtual_network.md), [azure_virtual_networks](docs/resources/azure_virtual_networks.md) |
 | azurerm_webapp, azurerm_webapps | `2016-08-01` | [azure_webapp](docs/resources/azure_webapp.md), [azure_webapps](docs/resources/azure_webapps.md) |
+| azure_bastion_hosts_resource, azure_bastion_hosts_resources | `2021-07-15` | [azure_bastion_hosts_resource](docs/resources/azure_bastion_hosts_resource.md), [azure_bastion_hosts_resources](docs/resources/azure_bastion_hosts_resources.md) |
```
Comment (sathish-progress) [inline README.md:395]: these are for legacy we add only resource with `azurerm_*`

File: docs/resources/azure_bastion_hosts_resource.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+end
+```
+## Parameters
+
+| Name                           | Description                                                                      |
+|--------------------------------|----------------------------------------------------------------------------------|
+| resource_group                 | Azure resource group that the targeted resource resides in. `MyResourceGroup`    |
+| name                           | Name of the Azure resource to test. `MyBastionHostName`                          |
+| type                           | type of BastionHostName                                                          |
+| provisioning_state             | State of BastionHostName creation                                                |
```
Comment (sathish-progress) [inline docs/resources/azure_bastion_hosts_resource.md:42]: are they parameters for this resource?

File: docs/resources/azure_bastion_hosts_resources.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+## Properties
+
+|Property       | Description                                                                          | Filter Criteria<superscript>*</superscript> |
+|---------------|--------------------------------------------------------------------------------------|-----------------|
+| name           | A list of the unique resource names.                                                | `name`            |
+| ids            | A list of bastion hosts ids .                                                       | `id`              |
+| tags           | A list of `tag:value` pairs defined on the resources.                               | `tag`             |
+| provisioning_state             | State of BastionHosts creation                                      | `provisioning_state`         |
+| types             |   Types of all the bastion hosts | `type` |
+| properties             |   Types of all the bastion hosts | `property` |
```
Comment (sathish-progress) [inline docs/resources/azure_bastion_hosts_resources.md:48]: field as described [here](https://github.com/inspec/inspec-azure/pull/414/files#diff-201aca2a0b9ac95395766f7e16f1328dd1fb4b328bee62705321a4667f51b444R63) should be `properties`

File: docs/resources/azure_bastion_hosts_resources.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+end
+```
+
+## Properties
+
+|Property       | Description                                                                          | Filter Criteria<superscript>*</superscript> |
+|---------------|--------------------------------------------------------------------------------------|-----------------|
+| name           | A list of the unique resource names.                                                | `name`            |
+| ids            | A list of bastion hosts ids .                                                       | `id`              |
+| tags           | A list of `tag:value` pairs defined on the resources.                               | `tag`             |
```
Comment (sathish-progress) [inline docs/resources/azure_bastion_hosts_resources.md:45]: field should be `tags` as described [here](https://github.com/inspec/inspec-azure/pull/414/files#diff-201aca2a0b9ac95395766f7e16f1328dd1fb4b328bee62705321a4667f51b444R60)

File: libraries/backend/azure_environment.rb (hunk 2 above)
Comment (sathish-progress) [reply libraries/backend/azure_environment.rb:101]: these changes are not relevant here actually

File: libraries/backend/helpers.rb [hunk 7]
```diff
@@ -92,6 +92,7 @@ class AzureEnvironments
       active_directory_endpoint_url: MicrosoftRestAzure::AzureEnvironments::AzureCloud.active_directory_endpoint_url,
       storage_endpoint_suffix: MicrosoftRestAzure::AzureEnvironments::AzureCloud.storage_endpoint_suffix,
       key_vault_dns_suffix: MicrosoftRestAzure::AzureEnvironments::AzureCloud.key_vault_dns_suffix,
+      datalake_store_filesystem_endpoint_suffix: MicrosoftRestAzure::AzureEnvironments::AzureCloud.datalake_store_filesystem_endpoint_suffix,
```
Comment (sathish-progress) [inline libraries/backend/helpers.rb:95]: all changes in this file are not relevant here again

File: libraries/backend/azure_environment.rb (hunk 2 above)
Comment (sa-progress) [reply libraries/backend/azure_environment.rb:101]: Yes Sateesh Already removed .If you see Master has same config.

File: libraries/backend/helpers.rb (hunk 7 above)
Comment (sa-progress) [reply libraries/backend/helpers.rb:95]: done.

File: libraries/backend/azure_environment.rb (hunk 2 above)
Comment (sa-progress) [reply libraries/backend/azure_environment.rb]: reverted from master.

File: docs/resources/azure_bastion_hosts_resource.md (hunk 4 above)
Comment (sa-progress) [reply docs/resources/azure_bastion_hosts_resource.md:54]: updated.

File: README.md (hunk 3 above)
Comment (sa-progress) [reply README.md]: updated.

File: libraries/azure_bastion_hosts_resource.rb [hunk 8]
```diff
@@ -0,0 +1,137 @@
+require 'azure_generic_resource'
+
+class AzureBastionHostsResource < AzureGenericResource
+  name 'azure_bastion_hosts_resource'
+  desc 'Azure Bastion to connect to a data lake hosts'
+  example <<-EXAMPLE
+    describe azure_bastion_hosts_resource(resource_group: 'example', bastionHostName: 'host-name') do
```
Comment (sathish-progress) [inline libraries/azure_bastion_hosts_resource.rb:7]: required parameters say `name` while you use `bastionHostName`

File: docs/resources/azure_bastion_hosts_resource.md [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+describe azure_bastion_hosts_resource(resource_group: 'MyResourceGroup', name: 'bastion_name') do
+  it { should exist }
+end
+```
+## Parameters
+
+| Name                           | Description                                                                      |
+|--------------------------------|----------------------------------------------------------------------------------|
+| resource_group                 | Azure resource group that the targeted resource resides in. `MyResourceGroup`     |
+| bastionHostName                           | Name of the virtual network to test. `MyVNetwork`                                 |
```
Comment (sathish-progress) [inline docs/resources/azure_bastion_hosts_resource.md:40]: in source code you use `name` as required parameter

File: docs/resources/azure_bastion_hosts_resource.md [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+
+Both of the parameter sets should be provided for a valid query:
+- `resource_group` and `bastionHostName`
+
+
+## Properties
+
+| Name                           | Description                                                                      |
+|--------------------------------|----------------------------------------------------------------------------------|
+| resource_group                 | Azure resource group that the targeted resource resides in. `MyResourceGroup`    |
```
Comment (sathish-progress) [inline docs/resources/azure_bastion_hosts_resource.md:50]: there is no resource group in the response

File: docs/resources/azure_bastion_hosts_resource.md [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+```
+## Parameters
+
+| Name                           | Description                                                                      |
+|--------------------------------|----------------------------------------------------------------------------------|
+| resource_group                 | Azure resource group that the targeted resource resides in. `MyResourceGroup`     |
+| name                           | Name of the virtual network to test. `MyVNetwork`                                 |
+
+Both of the parameter sets should be provided for a valid query:
+- `resource_group` and `bastionHostName`
```
Comment (sathish-progress) [inline docs/resources/azure_bastion_hosts_resource.md:43]: here too it should be `name`

File: docs/resources/azure_bastion_hosts_resource.md (hunk 9 above)
Comment (sa-progress) [reply docs/resources/azure_bastion_hosts_resource.md]: Yes.Corrected both places.

File: docs/resources/azure_bastion_hosts_resources.md [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+```
+
+## Properties
+
+|Property       | Description                                                                          | Filter Criteria<superscript>*</superscript> |
+|---------------|--------------------------------------------------------------------------------------|-----------------|
+| name           | A list of the unique resource names.                                                | `name`            |
+| ids            | A list of bastion hosts ids .                                                       | `id`              |
+| tags           | A list of `tag:value` pairs defined on the resources.                               | `tags`             |
+| provisioning_state             | State of BastionHosts creation                                      | `provisioning_state`         |
```
Comment (sathish-progress) [inline docs/resources/azure_bastion_hosts_resources.md:46]: Could you rename provisioning state as per your source code https://github.com/inspec/inspec-azure/pull/414/files#diff-201aca2a0b9ac95395766f7e16f1328dd1fb4b328bee62705321a4667f51b444R61

File: terraform/azure.tf [hunk 13]
```diff
@@ -1317,6 +1317,29 @@ resource "azurerm_policy_assignment" "inspec_compliance_policy_assignment" {
...
+  name                = "examplebastion"
+  location            = azurerm_resource_group.rg.location
+  resource_group_name = azurerm_resource_group.rg.name
+
+  ip_configuration {
+    name                 = "configuration"
+    subnet_id            = azurerm_subnet.subnet.id
+    public_ip_address_id = azurerm_public_ip.public_ip.id
+  }
+}
```
Comment (sathish-progress) [inline terraform/azure.tf:1331]: I guess this is a duplicate

**Review by sathish-progress — APPROVED** [review]
LGTM

**Pull request merged**